pkg/sessions/sql/sql_store.go

```diff
@@ -0,0 +1,92 @@
+package sql
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	// load the driver
+	_ "github.com/lib/pq"
+
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/options"
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/apis/sessions"
+
+	"github.com/oauth2-proxy/oauth2-proxy/v7/pkg/sessions/persistence"
+)
+
+// SessionStore is an implementation of the persistence.Store
+// interface that stores sessions in a SQL database
+type SessionStore struct {
+	pool  *sql.DB
+	table string
+}
+
+// NewSQLSessionStore initialises a new instance of the SessionStore and wraps
+// it in a persistence.Manager
+func NewSQLSessionStore(opts *options.SessionOptions, cookieOpts *options.Cookie) (sessions.SessionStore, error) {
+	table := opts.SQL.TablePrefix + "_sessions"
+	pool, err := sql.Open(opts.SQL.Driver, opts.SQL.DSN)
+	if err != nil {
+		return nil, fmt.Errorf("error constructing SQL client: %w", err)
+	}
+	_, err = pool.Exec(fmt.Sprintf(`CREATE TABLE IF NOT EXISTS %s (
```

Comment:
Should we do these SQL schema setup type operations in a `migrations` library?

That we we can cleanly manage changes we might make in a safe fashion during any version upgrades?

Reply:
Unsure if it's worth actually splitting this into a library if we also split off the client struct like you proposed in the other comment, but can do. I recognize that this bare `Exec` is quite primitive, but OTOH I didn't want a full ORM-like migration for a single table with three columns.